src/wp-includes/embed.php

```diff
@@ -1064,13 +1064,7 @@ function enqueue_embed_scripts() {
  * @since 4.4.0
  */
 function print_embed_styles() {
-	$type_attr = current_theme_supports( 'html5', 'style' ) ? '' : ' type="text/css"';
-	$suffix    = SCRIPT_DEBUG ? '' : '.min';
-	?>
-	<style<?php echo $type_attr; ?>>
-		<?php echo file_get_contents( ABSPATH . WPINC . "/css/wp-embed-template$suffix.css" ); ?>
-	</style>
-	<?php
+	wp_enqueue_style( 'wp-embed-template' );
```

Comment:
The `print_embed_styles()` function runs at the `embed_head` action. Since `embed_head` also runs `wp_print_styles` later at priority 20, this should work.

Reply:
I consider this approach problematic since it is strictly speaking a backward compatibility break: The function used to print styles, now it doesn't, it simply enqueues styles. A direct call to this function will not result in the expected outcome.

I think we should leave this function as is, deprecate it, and instead introduce `enqueue_embed_styles()` which does this to replace the current function.